### ISSUE TITLE
Using quoted keys in enum enforces array type for tuples

### ISSUE THREAD

**Issue by jeegurda** [microsoft/TypeScript#47737]
# Bug Report

Using quoted keys for enums (e.g. when they start with a number) weirdly breaks typing for objects.
I'm trying to specify tuples as object values but TS doesn't allow it: when using enum member with quotes, tuple value types are enforced as arrays. 

Please see the playground, there is a number of workarounds i've found which leads me to believe this is a bug.
I found [this issue](https://github.com/microsoft/TypeScript/issues/43171#issuecomment-795652146) which has a great explanation, but judging by the specified workarounds it doesn't seem to be the case here. 
Please note that my case fails for tuples specifically, but using other types works (provided below and on playground)

### 🔎 Search Terms

computed enum keys, enum keys, enum

### 🕗 Version & Regression Information

4.5.4
Same behavior in nightly (4.6.0)

### ⏯ Playground Link

[Playground link](https://www.typescriptlang.org/play?#code/KYOwrgtgBA0sCeBnKBvAsAKClAYgSQCUBlAFQBpNsiBRAYQHkA5AEUwF9NMB6LqEgCwBOAewDuyAIYgowQSMGYAxsJCIALlGEAjAFYAuVJSgBtANYIoAS2lwkAXQPH1g6wHMyUZ27vsoAXkMsE1tEADp8YhIHEwByAEYYjxiAJhi7CiDjEOMYmgYWNOicgGZEqBiAFjSoHihqOWFBAxJ4AAdgTQAzKGUIVrA1YAATKFaRdsE1eBjkADcJABswDstkGbUXEFdjOzLRfktFfitkEGENCURES1cQCS0FjrVhKCn28qcNtw8vLd3Q9icDC1ADqjVMEhEYBAIziUAAFGcoABHMDnYCIACUShU6k0ujiBnQmXM8CsNgQiCKv3cni+f18AWJ2CylPChFIRXiZRSaQyLJCoTyTGYXNKSSq6RqvAAPABaOVQZjCDFQUTgtX8UAycDQCDACBaWQnTztRSWTqWYZqyxqfjCAYotGDRCAjDcXhgwQQqEwqDJBGKS5qNzYjDKVQabQ6ZJEoxmCzWWCU6n02k0nwYNj+QICtkRTmOblJVLpePZXJ0EWFIvi8qSqCXEw0n5puzSqDyxXK1Xq72a7XzJYrZDAAAerQWh1tCzJQfU1uerzAk+Abo9UC9PodfuKCKG41abmXq6xOMj+J0xTjJMTFPsBhpOw7XaVKuQfdMNrtjbkEiQHh2qsMgTlO5oaG8HQQBIahHKq1idLIgiLm0HRSCMQbSEamgQLagxDIyubBPmHJREWCQlny5aUjkwoFHYYo8pK67ArwPYfhqoi2sc5xaoIryocgipIkhjSuhgkFQEQ-ASO8TJQLqBi6ka-HZgAPqgrxjmoj5plA6lQFowjCI8UhAhGeKdA6kz8DeLKkuSyYPlJMntIRzLEUg7KRNEKAKZABgAKwAAz6fynmILRVb0b5-kQAYcTBaFbDhaySBRfkzA1ppajaQYMRatMYUdkMK5gTBHSkogHhaI6tpQEM74gDEGjQWogz8VZ-FnKI1HpZWmXZRsyxQCxQKYEAA)

### 💻 Code

```ts
enum Keys {
  FIRST,
  SECOND
}

const obj: { // Shows correct type in a tooltip, evaluated to just what we need: { 0: [string, string], 1: [string, string] }
  [key in Keys]: [string, string] 
} = {
  [Keys.FIRST]: ['1', '2'], 
  [Keys['SECOND']]: ['3', '4'] // Error: Type of computed property's value is 'string[]', which is not assignable to type '[string, string]'.
}


// Using types other than tuples eliminates the problem
type Shape = { num: number } | { txt: string } | boolean

const fourth: {
  [key in Keys]: Shape
} = {
  [Keys.FIRST]: { num: 50 },
  [Keys['SECOND']]: { num: 100 },
  [Keys['SECOND']]: { txt: 'hey' }, // duplicate keys, but it doesn't matter for now
  [Keys['SECOND']]: true 
}
```

Interestingly i found the same behavior when using objects _as const_
```ts
const keysStoredAsObject = {
  'FIRST': 0,
  'SECOND': 1
} as const
type ObjectKeys = keyof typeof keysStoredAsObject

const finalObj: {
  [key in typeof keysStoredAsObject[ObjectKeys]]: [string, string]
} = {
  [keysStoredAsObject.FIRST]: ['1', '2'], // All good
  [keysStoredAsObject['SECOND']]: ['3', '4'] // Error
}
```

### 🙁 Actual behavior

Doesn't allow tuples as values, throws an incompatibility error for `string[]` and `[string, string]` types

### 🙂 Expected behavior

Allows to specify tuples as values to the object instead of plain arrays, as specified in its type definition

**Comment by Zzzen** [comment]
Duplicate of #29718

**Comment by jeegurda** [comment]
Seems fixed in 5.1.6
